tests/core/abi-utils/test_abi_utils.py

```diff
@@ -104,58 +485,583 @@ def test_fn_abi_to_4byte_selector(fn_abi, expected):
+    "abi_element",
+    (
+        ABI_FALLBACK,
+        ABI_RECEIVE,
+    ),
+)
+def test_get_input_names_raises_for_fallback_or_receive_abi(
+    abi_element: ABIElement,
+) -> None:
+    with pytest.raises(
+        ValueError,
+        match=f"Inputs not supported for function types 'fallback' or 'receive'."
+        f" Provided ABI type was '{abi_element['type']}'.",
+    ):
+        get_abi_input_names(abi_element)
+
+
+@pytest.mark.parametrize(
+    "abi_element,input_args",
+    [
+        (
+            ABI_EVENT_LOG_SINGLE_ARG,
+            [{"name": "arg0", "type": "uint256"}],
+        ),
+        (
+            ABI_EVENT_LOG_SINGLE_WITH_INDEX,
+            [{"name": "arg0", "type": "uint256", "indexed": True}],
+        ),
+        (
+            ABI_EVENT_LOG_MULTI_ARGS,
+            [
+                {"name": "arg0", "type": "uint256"},
+                {"name": "arg1", "type": "uint256"},
+                {"name": "arg2", "type": "uint256"},
+            ],
+        ),
+        (
+            ABI_EVENT_LOG_NO_ARG,
+            [],
+        ),
+        (
+            ABI_FUNCTION_SINGLE_ARG,
+            [{"name": "arg0", "type": "uint256"}],
+        ),
+        (
+            ABI_FUNCTION_TWO_NAMED_ARGS,
+            [{"name": "arg0", "type": "uint256"}, {"name": "arg1", "type": "uint256"}],
+        ),
+        (
+            ABI_FUNCTION_MULTI_TYPED_ARGS,
+            [{"name": "arg0", "type": "uint256"}, {"name": "arg1", "type": "bytes32"}],
+        ),
+        (
+            ABI_CONSTRUCTOR_WITH_INPUT,
+            [{"name": "inputs", "type": "str"}],
+        ),
+        (
+            ABI_ERROR_INVALID,
+            [{"name": "a", "type": "address"}, {"name": "b", "type": "bytes32"}],
+        ),
     ],
-}
+)
+def test_get_input_types_from_abi_element(
+    abi_element: ABIElement, input_args: Sequence[ABIElement]
+) -> None:
+    assert get_abi_input_types(abi_element) == [arg.get("type") for arg in input_args]
+
+
+@pytest.mark.parametrize(
+    "abi_element",
+    (
+        ABI_FALLBACK,
+        ABI_RECEIVE,
+    ),
+)
+def test_get_input_types_raises_for_fallback_or_receive_abi(
+    abi_element: ABIElement,
+) -> None:
+    with pytest.raises(
+        ValueError,
+        match=f"Inputs not supported for function types 'fallback' or 'receive'."
+        f" Provided ABI type was '{abi_element['type']}'.",
+    ):
+        get_abi_input_types(abi_element)
+
+
+@pytest.mark.parametrize(
+    "abi_element,outputs",
+    [
+        (
+            ABI_FUNCTION_SINGLE_ARG,
+            [{"name": "arg0", "type": "uint256"}],
+        ),
+        (
+            ABI_FUNCTION_TWO_NAMED_ARGS,
+            [{"name": "arg0", "type": "uint256"}, {"name": "arg1", "type": "uint256"}],
+        ),
+        (
+            ABI_FUNCTION_MULTI_TYPED_ARGS,
+            [{"name": "arg0", "type": "uint256"}, {"name": "arg1", "type": "bytes32"}],
+        ),
+        (
+            ABI_FUNCTION_TWO_UNNAMED_ARGS,
```

Comment:
Just a note that this test is the most crucial here, since all of the other examples have identical `inputs` and `outputs`. I wonder if we should vary those a bit to test more accurately but at least this test ensures us that we return outputs and not inputs.